scripts/TRN_TRN.R

```diff
@@ -0,0 +1,264 @@
+         # this next test is redundant because of circularity? default to TRUE?
+         trn2inreg1 = TRUE
+  ) |>
+  select(trn1, trn2, trn1inreg2, trn2inreg1, is_title_matched) |>
+  ungroup()
+
+# Integrate title matched TRNs
+trn_trn_titles_tidy <- trn_trn_results_tidy |>
+  mutate(is_title_matched = NA) |>
+  rows_upsert(trn_trn_titles, by = c("trn1", "trn2")) |>
+  rowwise() |> # added this rowwise()
+  mutate(registry1 = case_when(is.na(registry1) ~ which_registry(trn1),
+                               .default = registry1),
+         registry2 = case_when(is.na(registry2) ~ which_registry(trn2),
+                               .default = registry2)
+  )
+
+####################################################################################################################
+## Finally add matches from TRNs found in publications
+
+# Unnest TRNs in pubs table
+trn_trn_pubs_long <- publications |>
+  rename(trn1 = primary_IV_id) |>
+  mutate(trn_si_long = strsplit(trns_si, ";"),
+         trn_abs_long = strsplit(trns_abs, ";"),
+         trn_ft_long = strsplit(trns_ft, ";")) |> # Split trns_si by semicolon to get individual trial IDs
+  unnest_longer(trn_si_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_si_long != trn1 | is.na(trn_si_long)) |> # filter out self-references
+  unnest_longer(trn_abs_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_abs_long != trn1 | is.na(trn_abs_long)) |> # filter out self-references
+  unnest_longer(trn_ft_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_ft_long != trn1 | is.na(trn_ft_long)) |> # filter out self-references
+  select(trn1, contains("long"))
+
+# Identify unique TRN pairings from pubs fields, fill out boolean information to identify pub source of match
+trn_trn_pubs <- trn_trn_pubs_long |>
+  rowwise() |>
+  mutate(trn2 = list(unique(c(trn_si_long, trn_abs_long, trn_ft_long)) |> na.omit())) |>
+  unnest_longer(trn2) |>
+  group_by(trn1, trn2) |>
+  summarise(pub_si = if_else(any(trn2 == trn_si_long), TRUE, NA),
+            pub_abs = if_else(any(trn2 == trn_abs_long), TRUE, NA),
+            pub_ft = if_else(any(trn2 == trn_ft_long), TRUE, NA)) |>
+  rowwise() |>
+  mutate(trn1inreg2 = trn1 %in% trn_registries$trn2[trn_registries$trn1 == trn2],
+         trn2inreg1 = trn2 %in% trn_registries$trn2[trn_registries$trn1 == trn1]) |>
+  select(trn1, trn2, trn1inreg2, trn2inreg1, contains("pub")) |>
+  ungroup()
+
+# Integrate pub matches to larger match table
+trn_trn_pubs_tidy <- trn_trn_titles_tidy |>
+  mutate(pub_si = NA, pub_abs = NA, pub_ft = NA) |>
+  rows_upsert(trn_trn_pubs, by = c("trn1", "trn2")) |>
+  mutate(registry1 = case_when(is.na(registry1) ~ which_registry(trn1),
+                               .default = registry1),
+         registry2 = case_when(is.na(registry2) ~ which_registry(trn2),
+                               .default = registry2)) |>
+  select(trn1:trn2inreg1, pub_si, pub_abs, pub_ft,
```

Comment:
please replace the colon to avoid location based selection. you can also just relocate. `relocate(contains("pub"), .after = trn2inreg1)`